conformance/utils/kubernetes/helpers.go

```diff
@@ -0,0 +1,251 @@
+			}},
+		}}
+		HTTPRouteMustHaveParents(t, c, routeNN, parents, 60)
+	}
+
+	return gwAddr
+}
+
+// WaitForGatewayAddress waits until at least one IP Address has been set in the
+// status of the specified Gateway.
+func WaitForGatewayAddress(t *testing.T, client client.Client, gwName types.NamespacedName, seconds int) (string, error) {
+	t.Helper()
+
+	var ipAddr string
+	waitFor := time.Duration(seconds) * time.Second
+	waitErr := wait.PollImmediate(5*time.Second, waitFor, func() (bool, error) {
+		gw := &v1alpha2.Gateway{}
+		err := client.Get(context.TODO(), gwName, gw)
+		if err != nil {
+			t.Logf("error fetching Gateway: %v", err)
+			return false, fmt.Errorf("error fetching Gateway: %w", err)
+		}
+
+		// TODO: Support more than IPAddress
+		for _, address := range gw.Status.Addresses {
+			if address.Type != nil && *address.Type == v1alpha2.IPAddressType {
+				ipAddr = address.Value
+				return true, nil
+			}
+		}
+
+		return false, nil
+	})
+	require.NoErrorf(t, waitErr, "error waiting for Gateway to have at least one IP address in status")
+	return ipAddr, waitErr
+}
+
+// HTTPRouteMustHaveParents waits for the specified HTTPRoute to have parents
+// in status that match the expected parents. This will cause the test to halt
+// if the specified timeout is exceeded.
+func HTTPRouteMustHaveParents(t *testing.T, client client.Client, routeName types.NamespacedName, parents []v1alpha2.RouteParentStatus, seconds int) {
+	t.Helper()
+
+	var actual []v1alpha2.RouteParentStatus
+	waitFor := time.Duration(seconds) * time.Second
+	waitErr := wait.PollImmediate(5*time.Second, waitFor, func() (bool, error) {
+		route := &v1alpha2.HTTPRoute{}
+		err := client.Get(context.TODO(), routeName, route)
+		if err != nil {
+			return false, fmt.Errorf("error fetching HTTPRoute: %w", err)
+		}
+
+		actual = route.Status.Parents
+
+		return parentsMatch(t, parents, actual), nil
+	})
+	require.NoErrorf(t, waitErr, "error waiting for HTTPRoute to have parents matching expectations")
+}
+
+func parentsMatch(t *testing.T, expected, actual []v1alpha2.RouteParentStatus) bool {
+	if len(expected) != len(actual) {
+		t.Logf("Expected %d Route parents, got %d", len(expected), len(actual))
+		return false
+	}
+
+	// TODO(robscott): Allow for arbitrarily ordered parents
+	for i, eParent := range expected {
+		aParent := actual[i]
+		if aParent.ControllerName != eParent.ControllerName {
+			t.Logf("ControllerName doesn't match")
+			return false
+		}
+		if reflect.DeepEqual(aParent.ParentRef.Group, eParent.ParentRef.Group) {
+			t.Logf("ParentRef.Group doesn't match")
+			return false
+		}
+		if reflect.DeepEqual(aParent.ParentRef.Kind, eParent.ParentRef.Kind) {
+			t.Logf("ParentRef.Kind doesn't match")
+			return false
+		}
+		if aParent.ParentRef.Name != eParent.ParentRef.Name {
+			t.Logf("ParentRef.Name doesn't match")
+			return false
+		}
+		if aParent.ParentRef.Namespace != eParent.ParentRef.Namespace {
+			t.Logf("ParentRef.Namespace doesn't match")
```

Comment:
nit: helpful to log what values were actually received vs expected

Reply:
Updated, thanks!